src/Orleans.Transactions/State/TransactionQueue.cs

```diff
@@ -482,6 +482,10 @@ private async Task StorageWork()
                     {
                         // process all committable entries, adding storage events to the storage batch
                         CollectEventsForBatch(committableEntries);
+                        if (problemFlag != TransactionalStatus.Ok)
+                        {
+                            return;
+                        }
```

Comment:
Should this be calling `NotifyOfRestore()`?

Reply:
No.  The finally block will check the problemFlag and handle any issues.  If Notify needs be called again, the finally block will do so.